File: portal-kernel/src/com/liferay/portal/kernel/model/Permission.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.portal.kernel.model;

/**
 * @author Brian Wing Shun Chan
 * @see    com.liferay.portal.model.impl.PermissionImpl
 */
public interface Permission {

	public String getActionId();

	public String getName();

	public String getPrimKey();

	public int getScope();

	public void setActionId(String actionId);

	public void setName(String name);

	public void setPrimKey(String primKey);

	public void setScope(int scope);

}

File: workspaces/liferay-testray-workspace/client-extensions/liferay-testray-custom-element/src/core/Permission.ts
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {Action} from '../types';

type PermissionObj = {
	href: string;
	method: string;
};

type ObjectPermission = {
	delete?: PermissionObj;
	get?: PermissionObj;
	update?: PermissionObj;
};

export default class Permission {
	static filterActions(
		actions: Action[],
		permissions: ObjectPermission = {}
	) {
		return actions.filter((action) => {
			return typeof action.permission === 'boolean'
				? action.permission
				: action.permission
				? !!(permissions as any)[action?.permission?.toLowerCase()]
				: true;
		});
	}
}
